### PR TITLE
Groundwork

### PR DESCRIPTION
**Pull request by tmancey**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[submodule "deps/bat-native-rapidjson"]
+	path = deps/bat-native-rapidjson
+	url = git@github.com:brave-intl/bat-native-rapidjson.git
+[submodule "deps/bat-native-usermodel"]
+	path = deps/bat-native-usermodel
+	url = git@github.com:brave-intl/bat-native-usermodel.git
+[submodule "third_party/sqlite_modern_cpp"]
+	path = third_party/sqlite_modern_cpp
+	url = git@github.com:SqliteModernCpp/sqlite_modern_cpp.git
+[submodule "third_party/curl"]
```
[the remark this answers is not in the document]
Comment (tmancey) [reply .gitmodules]: Curl will only be used for isolated testing outside of brave-core etc.

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[submodule "deps/bat-native-rapidjson"]
+	path = deps/bat-native-rapidjson
+	url = git@github.com:brave-intl/bat-native-rapidjson.git
+[submodule "deps/bat-native-usermodel"]
+	path = deps/bat-native-usermodel
+	url = git@github.com:brave-intl/bat-native-usermodel.git
+[submodule "third_party/sqlite_modern_cpp"]
```
[the remark this answers is not in the document]
Comment (tmancey) [reply .gitmodules]: SQLite will only be used for isolated testing outside of brave-core etc. as each client will have their own implementation

File: .gitmodules (hunk 1 above)
[the remark this answers is not in the document]
Comment (bridiver) [reply .gitmodules]: 👍

File: .gitmodules (hunk 2 above)
[the remark this answers is not in the document]
Comment (bridiver) [reply .gitmodules]: 👍

[the remark this answers is not in the document]
Comment (bridiver) [reply .gitmodules]: also fyi - gn has a DEPS file that can pull these in and automatically update them

**Pull request merged**